x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportExecuteEnrichPolicyAction.java

```diff
@@ -58,7 +61,7 @@ public TransportExecuteEnrichPolicyAction(
         this.executor = new EnrichPolicyExecutor(
             settings,
             clusterService,
-            client,
+            new OriginSettingClient(client, ENRICH_ORIGIN),
```

Comment:
I don't know that this is correct. If I am not mistaken this is to take a user's policy and execute it to create the enrich index for the policy. The user's permissions dictate what indices they can access so we do not want to override that with the enrich origin when reading the data from the source index but instead need to do it only when writing the data to the enrich system indices unless I missed something?

Reply:
Looking at this closer, I think you're correct - I think I misread what this does the first time. Thanks for the catch!

(Also, do we not have enrich+security tests to catch something like this? I'll have to check)

Reply:
It looks like we do have some security tests for Enrich (`x-pack/plugin/enrich/qa/rest-with-security`) but it is a limited set of tests.

Reply:
We discussed this offline and came to the conclusion that this is indeed the incorrect way to handle this. I've updated this to ensure that all actions that touch the `source` index in an Enrich policy run in the user's security context, and the Enrich security context is used only for operations that only touch the `.enrich*` index.

This does run into a problem in the reindex call used to populate the `.enrich*` index, which touches both. In this case, we simply drop the response headers added during the reindex call - a change which will need a follow-up later, as that approach will not work when we enable full protection of system indices.